src/src/rx_main.cpp

```diff
@@ -1210,6 +1211,11 @@ void MspReceiveComplete()
 #endif
         break;
     case MSP_ELRS_MAVLINK_TLM: // 0xFD
+        if (config.GetSerialProtocol() != PROTOCOL_MAVLINK)
+        {
+            config.SetSerialProtocol(PROTOCOL_MAVLINK);
+            deferExecutionMicros(100, reconfigureSerial);
```

Comment:
Should be able to call `reconfigureSerial` directly as `MspReceiveComplete` is called from the main loop.

Reply:
Roger that, will test. 

This was just following the same method used in Lua.

Reply:
Thats 100us not ms, so thats not enough time for the eeprom write to happen!

Reply:
Thinking more after coffee, it should be delayed so this doesn't happen during the tryain isr.

Reply:
It can't happen in the ISR because `MspReceiveComplete` is only called from the main loop. Not in an ISR at any stage, so it should be completely safe.